sql/sys_vars.cc

```diff
@@ -6961,17 +6961,15 @@ static bool check_set_default_table_encryption_access(
   return true;
 }
 
-static const char *default_table_encryption_names[] = {
-    "OFF", "ON", "KEYRING_ON", "ONLINE_TO_KEYRING",
-    "ONLINE_FROM_KEYRING_TO_UNENCRYPTED"};
-
-bool Sys_var_enum_default_table_encryption::global_update(
-    THD *thd, set_var *var) {  // TODO: Test if it deos not overflow
-
-  if (thd->variables.default_table_encryption ==
-      var->save_result.ulonglong_value)
-    return 0;
-
+static const char *default_table_encryption_type_names[] = {
+    "OFF",
+    "ON",
+    "KEYRING_ON",
+    "ONLINE_TO_KEYRING",
+    "ONLINE_FROM_KEYRING_TO_UNENCRYPTED",
+    0};
```

Comment:
```suggestion
    nullptr};
```

That's the ASan fix right?

Reply:
I hope so - noticed it before you asked me about the ASAN issue, but it was most definitely causing out of scope memory accesses.